### PR TITLE
Connect redux and react

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Made sure that the state is immutable in all reducers
- Change the default state in the books reducer from an empty array to an array with a few books.
- Imported Redux Provider in the main component of the React app.
- Connected the app to the Redux store.
- Displayed books from Redux store.
- Added functionality for "Add new" button click.
- Added a functionality for "Remove" button click.
- Added a functionality for "Check status" button click.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 const initialState = {
-  categories: [],
+  categories: '',
 };
 
-export const checkCategoryStatus = (id) => ({ type: types.STATUS_CHECKED, id });
+export const checkCategoryStatus = () => ({ type: types.STATUS_CHECKED });
 
 export default function categoryReducer(state = initialState, action) {
   switch (action.type) {
```
Comment (zieeco) [inline src/redux/categories/categories.js:10]: - Please, kindly update the categories to return the 'Under construction' text as your's is returning an empty string. see below ::
```js
export default function categoryReducer(state = initialState, action) {
  switch (action.type) {
    case types.STATUS_CHECKED:
      return {
        ...state,
        categories: 'Under construction',
      };
    default:
      return state;
  }
}
```

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @JonahKayizzi ,


Great job so far 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 1 above)
Comment (JonahKayizzi) [reply src/redux/categories/categories.js:10]: Thanks for the observation. I will update accordingly

**Review by descholar-ceo — APPROVED** [review]
Hi @JonahKayizzi && @zieeco 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**